### PR TITLE
Update api and corresponding models

### PR DESCRIPTION
**Pull request by sambuddhabasu**

### CODE REVIEW

File: bookie/models/__init__.py [hunk 1]
```diff
@@ -184,8 +185,16 @@ def complete(prefix, current=None, limit=5, username=None):
...
 
             # if we have a username limit to only bookmarks of that user
-            if username is not None:
+            if username:
                 qry = qry.filter(Tag.bmark.any(username=username))
+                if username != requested_by:
+                    bmark = aliased(Bmark)
+                    qry = qry.join((bmark, Tag.bmark)).\
+                        filter(bmark.is_private == False)   # noqa
+            else:
```
Comment (craigmaloney) [inline bookie/models/__init__.py:198]: What happens if username == requested by? Looks like this falls through.

Comment (sambuddhabasu) [inline bookie/models/__init__.py:198]: if username == requested_by, we don't need to filter on anything, as in that case, we would want both the private and the public bookmarks. We already get both public and private bookmarks by the above line of code, qry = qry.filter(Tag.bmark.any(username=username))

File: bookie/tests/test_models/test_tagmgr.py [hunk 2]
```diff
@@ -24,27 +26,186 @@ def test_total_ct(self):
...
+
+    def test_basic_complete_same_user_accounts_for_privacy(self):
+        """Tags should provide completion options."""
+        user = User()
+        user.username = gen_random_word(10)
+        DBSession.add(user)
+
+        # Generate demo tag into the system
+        tags = [make_tag() for i in range(5)]
+        [DBSession.add(t) for t in tags]
```
Comment (mitechie) [inline bookie/tests/test_models/test_tagmgr.py:74]: I thought we were going to take these out and just pass the ' '.join(tags) into the BmarkMgr.store() ?

**Comment by mitechie** [comment]
One comment on making the tests simpler and then this looks good to go. Thanks for the hard work. :+1:

File: bookie/tests/test_models/test_tagmgr.py (hunk 2 above)
Comment (sambuddhabasu) [inline bookie/tests/test_models/test_tagmgr.py:74]: The BmarkMgr.store() has been replaced with make_bookmark() now, so we would need to add the tags.

**Comment by preddy5** [comment]
:+1:

**Comment by mitechie** [comment]
:+1:

**Comment by sambuddhabasu** [comment]
:shipit:

**Pull request closed without merging**